### PR TITLE
Added inclusion clause to files filter.

### PR DESCRIPTION
**Pull request by cdeveas**

### CODE REVIEW

File: lib/assetProcessor.js [hunk 1]
```diff
@@ -613,8 +613,13 @@ function _getRelevantFiles(root, directories, specificFiles, extensions, prefere
...
                 });
 
+                // check if the file is included in "preference"
+                var included = (preference || []).some(function(preference) {
+                    return file.toLowerCase() === preference.toLowerCase();
+                });
+
                 // if extension matches and not excluded, keep the file!
-                return extensionMatch && !excluded;
+                return extensionMatch && (!excluded || included);
```
Comment (Zugwalt) [inline lib/assetProcessor.js]: At this point maybe we should use a flag and progressively set to be better readable.
```
include = false

is it explicitly listed in preference?
    include = true
else does extension match?
   include = not excluded

return include;
```

File: package.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "name": "asset-processor",
   "version": "0.0.9",
   "scripts": {
-    "start": "node process.js"
+    "start": "node process.js",
+    "test": "nodeunit"
```
Comment (Zugwalt) [inline package.json]: nice!

File: test/test-assetProcessor.js [hunk 3]
```diff
@@ -54,6 +54,7 @@ exports.testGetJavaScriptFiles = function(test) {
                 'mixed/four.js',
                 'mixed/mixed_directory_1/five.js',
                 'mixed/mixed_directory_2/six.js',
+                'js/js_excluded/shouldBeIncluded.js',
```
Comment (Zugwalt) [inline test/test-assetProcessor.js]: nice

**Comment by Zugwalt** [comment]
Looks ok but would love to clean up the logic flow in assetProcessor.js if you can. Should be fast and safe now that you have a unit test to develop against!

**Comment by cdeveas** [comment]
Thanks for checking out. How would we re-publish this to npm?

**Comment by Zugwalt** [comment]
Right now I think I just have to do it using npm

File: lib/assetProcessor.js (hunk 1 above)
Comment (cdeveas) [inline lib/assetProcessor.js]: Cool will change.

**Pull request merged**